fe2-android/app/src/main/java/com/github/dedis/popstellar/model/network/method/message/data/federation/ChallengeRequest.kt

```diff
@@ -15,6 +16,10 @@ class ChallengeRequest
  */
 (val timestamp: Long) : Data {
 
+  init {
+    MessageValidator.verify().validPastTimes(timestamp)
```

Comment:
given the protocol : 

```
        "timestamp": {
            "type": "integer",
            "description": "[Timestamp] of the request",
            "minimum": 0
        }
```

timestamp should be >= 0, however `validPastTime `does not check this. I think that this check should be preceded by `greaterOrEqualThan`